### PR TITLE
Fix return tests

### PR DESCRIPTION
**Pull request by gianbelinche**
This PR fixes the tests on return.sol

For that I added some logic regarding far call reverts, specifically when an opcode inside a far call throws an error, however it is just the necessary stuff to make this tests pass, its not complete, and there could even be some unnecessary things.

### CODE REVIEW

**Review by jorbush — APPROVED** [review]
LGTM

File: src/lib.rs [hunk 1]
```diff
@@ -207,25 +207,37 @@ pub fn run(
...
+                            Err(should_break.unwrap_err())
+                        } else {
+                            if should_break.unwrap() {
+                                return Ok((ExecutionOutput::Panic, vm));
+                            }
+                            Ok(())
                         }
-                        Ok(())
                     }
                 },
```
Comment (juan518munoz) [inline src/lib.rs:242]: Simplify by `match`ing:
```rs
                Variant::Ret(ret_variant) => match ret_variant {
                    RetOpcode::Ok => match ok(&mut vm, &opcode) {
                        Ok(should_break) => {
                            if should_break {
                                break;
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    },
                    RetOpcode::Revert => match revert(&mut vm, &opcode) {
                        Ok(should_break) => {
                            if should_break {
                                return Ok((ExecutionOutput::Revert(vec![]), vm));
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    },
                    RetOpcode::Panic => match panic(&mut vm, &opcode) {
                        Ok(should_break) => {
                            if should_break {
                                return Ok((ExecutionOutput::Panic, vm));
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    },
                },
```

**Review by juan518munoz — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (juan518munoz) [reply src/lib.rs:242]: Even simpler (maybe not that legible?):
```rs
                Variant::Ret(ret_variant) => {
                    let result = match ret_variant {
                        RetOpcode::Ok => ok(&mut vm, &opcode),
                        RetOpcode::Revert => revert(&mut vm, &opcode),
                        RetOpcode::Panic => panic(&mut vm, &opcode),
                    };
                    match result {
                        Ok(should_break) => {
                            if should_break {
                                break;
                            }
                            Ok(())
                        }
                        Err(e) => Err(e),
                    }
                }
```

Comment (gianbelinche) [reply src/lib.rs:242]: The second one would not work, since the return values are different, I'll do the first one, thanks!

Comment (juan518munoz) [reply src/lib.rs]: You are right, missed it!

**Review by juan518munoz — APPROVED** [review]
LGTM

**Pull request merged**